### DISCUSSION TITLE
Access to record within custom validate method

### DISCUSSION THREAD

**Discussion by Daneng66** [react-hook-form/react-hook-form#7534] [Ideas] [answered]
It would be very handy to have access to the record as a prop of the custom validate method as part of register. This means you can do more complex cross field validation

**Comment by bluebill1049** [comment, marked as answer]
you can use `getValues()` to access other input values.

**Reply by Daneng66** [reply]
But within a field array you don't have access to the index so can't find the correct record after doing a getValues()

**Reply by bluebill1049** [reply]
why not pass in the index

{...register('test.0.data', { validate: () => {
  // index you can still access it here try console.log
}})}

**Reply by Moshyfawn** [reply]
See [this codesandbox example](https://codesandbox.io/s/use-field-array-index-validate-8yr75) on how to use field array index inside the register validation

**Reply by Daneng66** [reply]
Cheers guys that’s helped a lot

**Reply by bluebill1049** [reply]
thank you @Moshyfawn 🙏
